### PR TITLE
add support for postgresql via psycopg2

### PR DESCRIPTION
**Pull request by bloodearnest**
This PR adds optional integration with psycopg2, in the form of a custom connection class

If used, it provides

a) add sql queries to sentry reports
b) slow queries application logs

It also extends the test django and flask apps to use a postgresql db, and adds a setuptools extra for installing postgresql dependencies. We also drop support for pypy, as that requires a fair bit more work, see #84.

### CODE REVIEW

File: talisker/postgresql.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        query = prettify_sql(query)
+        query = FILTERED if query is None else query
+        duration = '{:d}ms'.format(int(math.ceil(t)))
+        connection = get_safe_connection_string(self)
+        return query, duration, connection
+
+    def _record(self, msg, query, duration):
+        query_data = None
+
+        if self.query_mintime >= 0 and duration > self.query_mintime:
```
Comment (ricardokirkner) [inline talisker/postgresql.py]: given this usage, should the variable not be best called query_maxtime?

File: talisker/postgresql.py [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+
+    @property
+    def query_mintime(self):
+        if self._mintime is None:
+            self._mintime = talisker.get_config()['slowquery_time']
+        return self._mintime
+
+    def cursor(self, *args, **kwargs):
+        kwargs.setdefault('cursor_factory', TaliskerCursor)
+        return super().cursor(*args, **kwargs)
```
Comment (ricardokirkner) [inline talisker/postgresql.py]: can you use super() in Python 2.7?

File: tests/django_app/manage.py [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!../../env/bin/python
```
Comment (ricardokirkner) [inline tests/django_app/manage.py]: possibly not for this PR but wouldn't it be better to move the django_app outside of the tests folder?

File: talisker/postgresql.py (hunk 1 above)
Comment (bloodearnest) [inline talisker/postgresql.py]: Hmm, perhaps, I don't mind either way, so I will rename

File: talisker/postgresql.py (hunk 2 above)
Comment (bloodearnest) [inline talisker/postgresql.py]: You can with python-future.org :) It's a bit lazy of me, as being explicit is supported in both, but I am so. very. tired. of explicit super in py2, so I just go all out and rely future's py2 workaround.

File: tests/django_app/manage.py (hunk 3 above)
Comment (bloodearnest) [inline tests/django_app/manage.py]: perhaps, but where would it live? It (and related flask/wsgi/celery apps) would need to be in another top level dir, not sure I see value in that.

FWIW, I explicitly exclude it from the test search path in pytest (see setup.cfg/norecursedirs)

File: talisker/postgresql.py (hunk 2 above)
Comment (ricardokirkner) [inline talisker/postgresql.py]: nice... lazyness is good

**Pull request merged**